packages/code-editor/src/CodeEditor/hooks/useLazyModules.ts

```diff
@@ -51,38 +106,29 @@ export type LoadersMap<T> = { [K in keyof T]: LoaderFn<T[K]> };
 export const useLazyModules = <T extends object>(
   loaders: LoadersMap<Partial<T>>,
 ) => {
-  const [modules, setModules] = useState<Partial<T>>({});
-  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [state, dispatch] = useReducer(lazyModulesReducer<T>, {
```

Comment:
nit: I usually destructure the state here for readability 
```suggestion
  const [{
    isLoading,
    keysToLoad,
    modules
  }, dispatch] = useReducer(lazyModulesReducer<T>, {
```